protocol/xmrmaker/swap_state.go

```diff
@@ -133,12 +133,14 @@ func newSwapStateFromStart(
 		offerExtra.StatusCh,
 	)
 
-	if err := b.SwapManager().AddSwap(info); err != nil {
+	if err = b.SwapManager().AddSwap(info); err != nil {
 		return nil, err
 	}
 
-	offerExtra.StatusCh <- stage
-	return newSwapState(
+	defer func() {
+		offerExtra.StatusCh <- stage
+	}()
```

Comment:
Should this be at the end of the function after we've cleared all the error scenarios, or is it okay here?